src/pybind/mgr/dashboard/controllers/prometheus.py

```diff
@@ -59,40 +92,64 @@ def alert_proxy(self, method, path, params=None, payload=None):
                                method, path, 'Alertmanager', params, payload,
                                user=user, password=password, verify=verify,
                                cert=cert, is_alertmanager=True)
-        self.close_unlink_files([ca_cert_file, cert_file, key_file])
         return response
 
     def get_access_info(self, module_name):
         # type (str, str, str, str, str)
-
-        def write_to_tmp_file(content):
-            # type (str)
-            if content is None:
-                return None
-            tmp_file = tempfile.NamedTemporaryFile(delete=False)
-            tmp_file.write(content.encode('utf-8'))
-            tmp_file.flush()  # tmp_file must not be gc'ed
-            return tmp_file
-
-        if module_name not in ['prometheus', 'alertmanager']:
-            raise DashboardException(f'Invalid module name {module_name}', component='prometheus')
+        orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
+        is_cephadm = orch_backend == 'cephadm'
         user = None
         password = None
         cert_file = None
         pkey_file = None
         ca_cert_file = None
-        orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
-        if orch_backend == 'cephadm':
-            cmd = {'prefix': f'orch {module_name} get-credentials'}
-            ret, out, _ = mgr.mon_command(cmd)
-            if ret == 0 and out is not None:
-                access_info = json.loads(out)
-                if access_info:
-                    user = access_info['user']
-                    password = access_info['password']
-                    ca_cert_file = write_to_tmp_file(access_info['certificate'])
-                    cert_file = write_to_tmp_file(mgr.get_localized_store("crt"))
-                    pkey_file = write_to_tmp_file(mgr.get_localized_store("key"))
+        secure_monitoring_stack = False
+        if is_cephadm:
+            def write_to_tmp_file(content):
+                # type (str)
+                if content is None:
+                    return None
+                tmp_file = tempfile.NamedTemporaryFile(delete=False)
+                tmp_file.write(content.encode('utf-8'))
+                tmp_file.flush()  # tmp_file must not be gc'ed
+                tmp_file.close()
+                return tmp_file
+
+            if module_name not in ['prometheus', 'alertmanager']:
+                raise DashboardException(f'Invalid module name {module_name}',
+                                         coFalsemponent='prometheus')
```

Comment:
this check previously failed regardless of the orchestrator, I think we should keep that way and fail as early as possible in case of bad module-name